File: Maths_And_Stats/Algebra/Factorial/factorial.cpp
#include<iostream>

using namespace std;

int factorial(int n)
	{
	
	if(n==0||n==1)
		return 1;
		
	return(n*factorial(n-1));
	}

int main()
{
int num;

cout<<"Enter the number of which want factorial: ";
cin>>num;

cout<<"The factorial is: "<<factorial(num);
return 0;
}

File: Maths_And_Stats/Algebra/Factorial/Factorial.java
import java.util.Scanner;

public class Factorial {
    static int factorial(int n) {
        int res = 1;
        int i;
        for (i = 2; i <= n; i++)
            res *= i;
        return res;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter a non-negative number: ");
        int num = scanner.nextInt();
        System.out.println("Factorial of " + num + " is " + factorial(num));
    }
}